src/components/bookmark-button/index.tsx

```diff
@@ -1,33 +1,38 @@
+import { useBookmarkMutations, useIsBookmarked } from 'hooks/bookmarks'
 import { AUTH_ENABLED } from 'hooks/use-authentication'
 import { RemoveBookmarkIcon, AddBookmarkIcon } from './icons'
 import s from './bookmark-button.module.css'
-
-/**
- * TODO
- * - create a 'plugged' in HOC that can render modals, update bookmark state etc.
- * This HOC will be used by this button and the one in tutorial meta
- */
+import { ApiTutorial } from 'lib/learn-client/api/api-types'
 
 interface BookmarkButtonProps {
-	isBookmarked: boolean
+	tutorialId: ApiTutorial['id']
 }
 
-export default function BookmarkButton({ isBookmarked }: BookmarkButtonProps) {
+export default function BookmarkButton({ tutorialId }: BookmarkButtonProps) {
+	const { isBookmarked } = useIsBookmarked({ tutorialId })
+	const { addBookmark, removeBookmark } = useBookmarkMutations()
```

Comment:
Do you mind if we do this integration in another PR? Just to avoid conflicts with some refactoring work I'm doing [in this PR](https://github.com/hashicorp/dev-portal/pull/837)

Reply:
Once this is merged I can integrate these updates into that PR

Reply:
Sure

Reply:
Actually on second thought, I'm not sure these changes should be undone. They are being used to test the new hooks.

Reply:
Once were done testing, we could undo? Or I can just manage the conflicts...

Reply:
I already had to deal with managing conflicts on this branch after spending many hours on it, so I'd appreciate it ❤️